flink-runtime/src/test/java/org/apache/flink/runtime/util/JarFileCreatorTest.java

```diff
@@ -218,7 +224,7 @@ public void TestAnonymousClass() throws IOException {
 
 	@Test
 	public void TestExtendIdentifier() throws IOException {
-		File out = new File(System.getProperty("java.io.tmpdir"), "jarcreatortest.jar");
+		File out = tempFolder.newFile("jarcreatortest.jar");
```

Comment:
to preserve the existing behavior of passing a file that does not exist, do this instead:
```
File out = new File(tempFolder.getRoot(), "jarcreatortest.jar");
```

Also applies to the other tests in this file.